src/ts/Conviva.d.ts

```diff
@@ -146,10 +146,14 @@ declare namespace Conviva {
     }
   }
 
+  type DeviceMetadata = {
```

Comment:
Why do we redefine a rather weak type here (Why do we need the `undefined`, btw - should be avoided in types unless really necessary)? We have this properly defined in an interface in `ConvivaAnalytics.ts`. If we'd use that, the user would have a better type checking and potentially auto-completion to create the metadata object.

Reply:
If you see where this type is used and the comparison with previous changes you will see that they are pretty similar. The difference is supporting other types for the values of this object because with new device metadata options can have other types.
Regarding the use of `undefined` it could be deleted.